cassandane/Cassandane/Cassini.pm

```diff
@@ -138,10 +138,24 @@ sub instance
 
 sub val
 {
-    # Args are: section, name, default
+    my ($self, $section, $name, $default) = @_;
+
+    # Allow overrides from specially-named environment variables.
+    #
+    # Examples:
+    #
+    # to override the "rootdir" option from the "[cassandane]" section,
+    # set: CASSINI_CASSANDANE_ROOTDIR=/some/different/value
+    #
+    # to override the "prefix" option from the "[cyrus default]" section,
+    # set: CASSINI_CYRUS_DEFAULT_PREFIX=/some/different/value
+    #
+    my $envname = "\UCASSINI $section $name\E";
+    $envname =~ s{[^A-Z0-9]+}{_}g;
+    return $ENV{$envname} if defined $ENV{$envname};
```

Comment:
I wonder if we should write a log that we found an environment variable that overrides the config? It looks as if using an environment variable is more of the exception.

Reply:
Yeah good thought!

Reply:
Added a log message.  Conveniently, you can see what it looks like from the new test:

```
./testrunner.pl -vvv -fpretty Cassini.environment_override
[20409] =====> Cassini::new[102] Reading /home/ellie/cassandane.ini
[20411] =====> Test::Cassini::test_environment_override[186] Working in temporary directory /tmp/3kz6Smui1L
[20411] =====> Cassini::new[102] Reading /tmp/3kz6Smui1L/cassandane.ini
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_ROOTDIR="overridden!"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_ROOTDIR=""
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CYRUS_DEFAULT_PREFIX="overridden!"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CYRUS_DEFAULT_PREFIX=""
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP="no"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP="NO"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP="false"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP="FALSE"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP="0"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP="off"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP="OFF"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP="yes"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP="YES"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP="true"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP="TRUE"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP="1"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP="on"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP="ON"
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP=""
[20411] =====> Cassini::val[156] Using configuration from environment: $CASSINI_CASSANDANE_CLEANUP="invalid"
Test::Cassini.environment_override                                      [  OK  ]

Time:  0 wallclock secs ( 0.01 usr  0.02 sys +  0.13 cusr  0.08 csys =  0.24 CPU)

OK (1 tests)
```